File: src/ddf/condition.py
from .base import DictMixin


class Condition(DictMixin):
    """Base condition."""


class ValueIs(Condition):
    """Rule that passes if a field has an arbitrary value."""

    def __init__(self, field, value):
        self.field = field
        self.value = value

    def validate(self, form):
        return form.data.get(self.field, None) == self.value

File: src/ddf/condition.js
// Validate if a field has a given value.
class ValueIs {
  constructor(field, value) {
    this.field = field
    this.value = value
  }

  // Return true if the field's value is this.value
  validate(form) {
    let valid = form.fieldValueGet(this.field) == this.value
    if (ddf.debug) console.log('[Condition]', this, '.validate(', form, ') = ', valid)
    return valid
  }
}

export {
  ValueIs,
}
